src/swap/components/SwapAmountInput.tsx

```diff
@@ -0,0 +1,57 @@
+import { useCallback, CSSProperties, useEffect } from 'react';
+
+import { SwapAmountInputReact } from '../types';
+import { isValidAmount } from '../utils';
+import { TokenSelector, TokenSelectorDropdown } from '../../token';
+
+export function SwapAmountInput({
+  token,
+  swappableTokens,
+  label,
+  amount,
+  tokenBalance,
+  onAmountChange,
+  onTokenSelectorClick,
+  onMaxButtonClick,
+  disabled = false,
+}: SwapAmountInputReact) {
+  const handleAmountChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      if (isValidAmount(event.target.value)) {
+        onAmountChange(event.target.value);
+      }
+    },
+    [onAmountChange],
+  );
+
+  return (
+    <div data-testid="ockSwapAmountInput_Container" className="ock-swapamountinput-container">
+      <div className="ock-swapamountinput-row">
+        <label className="ock-swapamountinput-label">{label}</label>
+        {tokenBalance && (
+          <label className="ock-swapamountinput-balance">{`Balance: ${tokenBalance}`}</label>
+        )}
+      </div>
+      <div className="ock-swapamountinput-row">
+        <TokenSelector token={token} setToken={onTokenSelectorClick}>
+          <TokenSelectorDropdown setToken={onTokenSelectorClick} options={swappableTokens} />
+        </TokenSelector>
+        <button
+          data-testid="ockSwapAmountInput_MaxButton"
+          className="ock-swapamountinput-maxbutton"
+          onClick={onMaxButtonClick}
```

Comment:
I'm wondering if it would be better to create a callback function within this component `handleMaxButtonClick` which calls `onAmountChange` with the `tokenBalance`.

Reply:
I agree with that approach!

Reply:
updated